test/e2e-tests.sh

```diff
@@ -121,6 +121,23 @@ go_test_e2e -timeout=2m ./test/e2e/initcontainers ${TEST_OPTIONS} || failed=1
 toggle_feature kubernetes.podspec-init-containers Disabled
 toggle_feature kubernetes.podspec-volumes-emptydir Disabled
 
+# RUN PVC tests with default storage class if available.
+if (( USE_DEFAULT_STORAGE )) ; then
+  kubectl apply -f "${E2E_YAML_DIR}/test/config/pvc/pvc.yaml" -n serving-tests
+fi
+
+toggle_feature kubernetes.podspec-persistent-volume-claim Enabled
+toggle_feature kubernetes.podspec-persistent-volume-write Enabled
+toggle_feature kubernetes.podspec-securitycontext Enabled
+go_test_e2e -timeout=5m ./test/e2e/pvc ${TEST_OPTIONS} || failed=1
```

Comment:
Will this work even if USE_DEFAULT_STORAGE returns false and the pvc.yaml is not applied? Just curious if this whole section should be within `if (( USE_DEFAULT_STORAGE )) ; then`

Reply:
Afaik most cloud providers and local setups have a default storage class, including:
- Kind (hostpath based)
- Minikube (hostpath based)
- [AWS](https://docs.aws.amazon.com/eks/latest/userguide/storage-classes.html)
- [Azure](https://docs.microsoft.com/en-us/azure/azure-arc/data/storage-configuration)
- [GCP](https://cloud.google.com/kubernetes-engine/docs/concepts/persistent-volumes#storageclasses)
My assumption is that if someone sets `USE_DEFAULT_STORAGE=0` then he knows what he is doing because he should then provide his own storage class to make tests run. By allowing to disable the default pvc we allow tests to integrate with any cluster. I am not sure how likely is to run on some infra where nothing is provided. If we add the whole section under that flag then we restrict tests from running on a modified cluster. That was my rationale here.

Reply:
I still think you need the PVC applied otherwise the pod references a resource that doesn't exist. So I don't really see the need for `USE_DEFAULT_STORAGE`

Reply:
> I still think you need the PVC applied otherwise the pod references a resource that doesn't exist.
If the user provides the pvc, it does exist no?